app/models/follower.rb

```diff
@@ -0,0 +1,9 @@
+class Follower < ApplicationRecord
+  belongs_to :follower, class_name: "Author", foreign_key: "follower_username", primary_key: "username"
+  belongs_to :following, class_name: "Author", foreign_key: "following_username", primary_key: "username"
+
+  validates_uniqueness_of :follower_username, scope: :following_username
+
+  scope :followers_from_author, ->(author_id) { where(author_id: author_id) }
```

Comment:
`author_id` doesn't exists on your follower table so filtering by author_id would not make sense